pkg/logql/engine.go

```diff
@@ -168,6 +168,9 @@ type EngineOpts struct {
 	// Batch size of the v2 execution engine.
 	BatchSize int `yaml:"batch_size" category:"experimental"`
 
+	// MergePrefetchConcurrency controls the number of inputs that are prefetched simultaneously by any Merge node.
+	MergePrefetchConcurrency int `yaml:"merge_prefetch_concurrency" category:"experimental"`
```

Comment:
Is it time for us to move all the V2 config options into a struct in pkg/engine and squash it here so we can constrain where we need to make changes for the v2 engine?

Reply:
Sounds about the right time.

Reply:
Let's do that in a separate PR, ok?